Automattic/wp-calypso
# Domains: Create new "Add DNS record" page

### Changes proposed in this Pull Request

This PR creates a new "Add a new DNS record" page, accessible via the `/domains/manage/:domain/add-dns-record/:site` URL. It can also be accessed from the new "DNS records" page, via the "Add a new record" button in the breadcrumbs area. Originally, the add a new DNS record form was located in the same page where all DNS records for a domain were listed. This is part of the domain management pages redesign project detailed in pcYYhz-m2-p2.

#### Screenshots

This is the new page:

<img width="989" alt="Screen Shot 2021-11-08 at 19 53 49" src="https://user-images.githubusercontent.com/5324818/140832299-b80c3d18-6aee-4dd3-9076-0fe787d749f6.png">

New page in mobile view:

<img width="397" alt="Screen Shot 2021-11-08 at 20 06 55" src="https://user-images.githubusercontent.com/5324818/140832405-4d7e7fc0-1cbe-4318-b8e3-c4cc2ff2ac3c.png">

This is where this functionality was located before:

![Annotation on 2021-11-08 at 20-01-02](https://user-images.githubusercontent.com/5324818/140832417-fac6508c-ef18-4e8c-a1fb-63a8c1567900.png)

### Testing instructions

Access the live Calypso link, navigate to "Upgrades > Domains > Change your name servers and DNS records > DNS records", append the `?flags=domains/dns-records-redesign` flag to the URL and click on the "Add a new record" button. Alternatively, access the `/domains/manage/:domain/add-dns-record/:site` URL directly (the feature flag is not needed there because it's a new path).

Try adding some DNS records and ensure they are created correctly. Please also check the mobile view.
